### PR TITLE
Slightly improve error message for invalid identifier

### LINKED ISSUE #104088: Slightly improve error message for invalid identifer
Given the following code: [link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=28746db28d4d778b7165a86aeb1f80b3)

```rust
fn main() {
    let 1x = 123;
}
```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: invalid suffix `x` for number literal
 --> src/main.rs:2:9
  |
2 |     let 1x = 123;
  |         ^^ invalid suffix `x`
  |
  = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)

error: could not compile `playground` due to previous error
```

Ideally the output should inform the user that an identifier cannot start with a digit and not that the proceeding characters is an invalid suffix (since a literal in that place is invalid in the first place). Thanks.

### PR DESCRIPTION
**Pull request by chenyukang**
fixes #104088

### CODE REVIEW

File: compiler/rustc_parse/src/parser/stmt.rs [hunk 1]
```diff
@@ -353,6 +354,18 @@ impl<'a> Parser<'a> {
...
     }
 
+    /// report error for `let 1x = 123`
+    pub fn report_invalid_identifier_error(&mut self) -> PResult<'a, ()> {
+        if let token::Literal(lit) = self.token.uninterpolate().kind &&
+            let Err(_) = rustc_ast::Lit::from_token(&self.token) &&
+            (lit.kind == token::LitKind::Integer || lit.kind == token::LitKind::Float) &&
+            self.look_ahead(1, |t| matches!(t.kind, token::Eq) || matches!(t.kind, token::Colon ) ) {
+                let err = self.sess.create_err(InvalidIdentiferStartsWithNumber { span: self.token.span });
+                return Err(err);
```
Comment (davidtwco) [inline compiler/rustc_parse/src/parser/stmt.rs:364]: nit:
```suggestion
                return Err(self.sess.create_err(InvalidIdentiferStartsWithNumber { span: self.token.span }));
```

**Review by davidtwco — APPROVED** [review]
LGTM, one nit then r=me

File: compiler/rustc_parse/src/parser/stmt.rs (hunk 1 above)
Comment (chenyukang) [reply compiler/rustc_parse/src/parser/stmt.rs]: Updated, thanks!

**Review by davidtwco — APPROVED** [review]

**Pull request merged**